au/code/au/cpp20_test.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 Aurora Operations, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if __cplusplus >= 202002L
```

Comment:
@geoffviola, FYI, here's the reason for this.  It turns out that even though gcc 10 does support the needed C++ features, they list their version number as, I think, `201709L`.  Guarding the entire test file seemed like the cleanest approach to me, because "these are the tests for people using C++20".  I think it's enough as long as we have _some_ configurations that are running these tests --- we don't need the whole cartesian product --- and I verified that this is in fact the case.